manim/utils/iterables.py

```diff
@@ -103,57 +121,72 @@ def batch_by_property(
     return batch_prop_pairs
 
 
-def concatenate_lists(*list_of_lists: Iterable) -> list:
+def concatenate_lists(*list_of_lists: Iterable[T]) -> list[T]:
     """Combines the Iterables provided as arguments into one list.
 
     Examples
     --------
-    Normal usage::
+    .. code-block:: pycon
 
-        concatenate_lists([1, 2], [3, 4], [5])
-        # returns [1, 2, 3, 4, 5]
+        >>> concatenate_lists([1, 2], [3, 4], [5])
+        [1, 2, 3, 4, 5]
     """
     return [item for lst in list_of_lists for item in lst]
 
 
-def list_difference_update(l1: Iterable, l2: Iterable) -> list:
+def list_difference_update(l1: Iterable[T], l2: Iterable[T]) -> list[T]:
     """Returns a list containing all the elements of l1 not in l2.
 
     Examples
     --------
-    Normal usage::
+    .. code-block:: pycon
 
-        list_difference_update([1, 2, 3, 4], [2, 4])
-        # returns [1, 3]
+        >>> list_difference_update([1, 2, 3, 4], [2, 4])
+        [1, 3]
     """
     return [e for e in l1 if e not in l2]
 
 
-def list_update(l1: Iterable, l2: Iterable) -> list:
+def list_update(l1: Iterable[T], l2: Iterable[T]) -> list[T]:
     """Used instead of ``set.update()`` to maintain order,
         making sure duplicates are removed from l1, not l2.
         Removes overlap of l1 and l2 and then concatenates l2 unchanged.
 
     Examples
     --------
-    Normal usage::
+    .. code-block:: pycon
 
-        list_update([1, 2, 3], [2, 4, 4])
-        # returns [1, 3, 2, 4, 4]
+        >>> list_update([1, 2, 3], [2, 4, 4])
+        [1, 3, 2, 4, 4]
     """
     return [e for e in l1 if e not in l2] + list(l2)
 
 
-def listify(obj) -> list:
+@overload
+def listify(obj: str) -> list[str]: ...
+
+
+@overload
+def listify(obj: Iterable[T]) -> list[T]: ...
+
+
+@overload
+def listify(obj: T) -> list[T]: ...
+
+
+def listify(obj):
```

Comment:
```suggestion
def listify(obj: str | Iterable[T] | T) -> list[str] | list[T]:
```

So that the function body is also type checked. You'll probably need to switch the body to:

```python
    if isinstance(obj, str):
        return [obj]
    if isinstance(obj, Iterable):
        return list(obj)
    return [obj]
```

So that it passes type checking (`try..expect` doesn't play well with type checkers).